### PR TITLE
Implement relations

### PR DESCRIPTION
**Pull request by DevAlien**
Still in draft but it is working, I need to clean up the code a bit.

Basically this one supports 2 kinds of relations:
- single relation (rel: "id")
- multiple relations (rel: [id, id])

and they get expanded when asked for, with metadata and everything `_relation=field`

image

### CODE REVIEW

File: dsi/postgres/project_resources.go [hunk 1]
```diff
@@ -681,6 +724,172 @@ func (d *Database) GetDefDocument(projectID, path, documentID string, filter map
...
+		if value, ok := obj[key]; ok {
+			switch value.(type) {
+			case []interface{}:
+				ss := make([]string, len(value.([]interface{})))
+				for i, v := range value.([]interface{}) {
+					ss[i] = fmt.Sprint(v)
+				}
+				documentIDs = append(documentIDs, ss...)
+			default:
+				documentIDs = append(documentIDs, value.(string))
```
Comment (anothrNick) [inline dsi/postgres/project_resources.go:786]: Shouldn't this always be a single string for `prepareRelationsSingle`?

File: dsi/postgres/project_resources.go [hunk 2]
```diff
@@ -681,6 +724,172 @@ func (d *Database) GetDefDocument(projectID, path, documentID string, filter map
...
+				for _, s := range value.([]interface{}) {
+					if relationValue, ok := relationObjs[s.(string)]; ok {
+						arrayV = append(arrayV, relationValue)
+					}
+				}
+				obj[key] = arrayV
+			default:
+				if relationValue, ok := relationObjs[value.(string)]; ok {
+					obj[key] = relationValue
+				}
```
Comment (anothrNick) [inline dsi/postgres/project_resources.go:810]: Same comment here

**Comment by anothrNick** [comment]
Previous comments resolved in Slack.

@DevAlien What's the status of this? It's marked as a WIP so I can't merge it.

**Comment by DevAlien** [comment]
Moved it, the code is not so great, but it works. I need to look on how we can add this to the documentation (haven't touched that project yet)

**Pull request merged**